### PR TITLE
Secure supply chain tests.

### LINKED ISSUE #19: Feature Request: Sigstore verification tests
### Is your feature request related to a problem?

_No response_

### Solution you'd like

Add tests for:

configure helm chart with verification enabled for kubewarden policies, but misconfigured
deploy policies, seeing that they are rejected
upgrade helm chart with correct verification configuration
configure helm chart with verification enabled for kubewarden policies
deploy policies, seeing that they are verified and instantiated

### Alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jvanz**
## Description

Adds Secure supply chain tests. 

Depends on  https://github.com/kubewarden/helm-charts/pull/75

Fix #19 

## Test

```shell
make create-k8s-cluster install-kubewarden secure-supply-chain-test
```

### CODE REVIEW

File: tests/secure-supply-chain-tests.bats [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	clean_up_environment
+	kubectl_delete_configmap_by_name $SECURE_SUPPLY_CHAIN_VERIFICATION_CONFIG_MAP_NAME
+}
+
+@test "[Secure Supply Chain tests] Trusted policy should not block policy server" {
+	create_configmap_from_file_with_root_key $SECURE_SUPPLY_CHAIN_VERIFICATION_CONFIG_MAP_NAME verification-config $RESOURCES_DIR/secure-supply-chain-verification-config.yaml
+	helm upgrade --set policyServer.verificationConfig=$SECURE_SUPPLY_CHAIN_VERIFICATION_CONFIG_MAP_NAME --wait -n kubewarden kubewarden-defaults kubewarden/kubewarden-defaults
+	kubectl_apply_should_succeed $RESOURCES_DIR/privileged-pod-policy.yaml
+	wait_for_all_cluster_admission_policies_to_be_active
+	wait_for_default_policy_server_rollout
```
Comment (jvanz) [inline tests/secure-supply-chain-tests.bats:23]: I'm aware that we have a [function](https://github.com/kubewarden/kubewarden-end-to-end-tests/blob/main/tests/common.bash#L23) calling similar functions. But in my tests, the function is not working properly and I need to double check why. I believe is an issue in the function used to check policy condition. I'll investigate that and update this PR or open another one to fix the issue.

File: resources/restricted-secure-supply-chain-verification-config.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# With this config, the only valid policies are those signed by Kubewarden
+# infrastructure.
+#
+# This config can be saved to its default location (for this OS) with:
+#   kwctl scaffold verification-config > /home/jvanz/.config/kubewarden/verification-config.yml
+#
+# Providing a config in the default location enables Sigstore verification.
+# See https://docs.kubewarden.io for more Sigstore verification options.
+---
+apiVersion: v1
```
Comment (raulcabello) [inline resources/restricted-secure-supply-chain-verification-config.yaml:12]: I think verification should not pass with this config unless I'm missing something here. I think it should be something like:

```
allOf:
  - kind: genericIssuer
    issuer: https://token.actions.githubusercontent.com
    subject:
      urlPrefix: https://github.com/kubewarden
 
```

Comment (raulcabello) [reply resources/restricted-secure-supply-chain-verification-config.yaml:12]: ignore this, I just realized this is the one that should fail :sweat_smile:

**Review by raulcabello — APPROVED** [review]
LGTM, thanks @jvanz !

**Review by viccuad — APPROVED** [review]
LGTM, thanks!

**Review by viccuad — APPROVED** [review]

**Review by ereslibre — APPROVED** [review]

**Review by raulcabello — APPROVED** [review]
LGTM

**Pull request merged**